### PR TITLE
Source Hubspot: Add OwnersArchived stream

### LINKED ISSUE #27799: Source HubSpot: Missing archived owners in Hubspot connector
### Connector Name

Hubspot

### Connector Version

latest

### What step the error happened?

Other

### Revelant information

We are using the Hubspot connector and realised that we need to include archived records when retrieving the Owners. Hubspot API by default do not include this so I asume a new flag needs to added in order to be able to include those.
I see you did it in other connectors like this one:

https://github.com/airbytehq/airbyte/pull/17371/files

Am I right and Owners is not including archived by default? 

This is the link to updated documentation from hubspot (There you can see that archived flag is FALSE by default):
https://developers.hubspot.com/docs/api/crm/owners

### Relevant log output

_No response_

### Contribute

- [ ] Yes, I want to contribute

### PR DESCRIPTION
**Pull request by arsenlosenko**
## What
Resolve issue described here:
https://github.com/airbytehq/airbyte/issues/27799

## How
Right now stream `Owners` returns records that are not archived, in order to cover this case i have added `OwnersArchived` stream which filters records and returns only archived records.

### CODE REVIEW

**Comment by octavia-squidington-iii** [comment]
##  source-hubspot test report (commit [`9e21e84bea`](https://github.com/airbytehq/airbyte/pull/29249/commits/9e21e84bea6764a695bb6c0804bcfcd166b35141)) - ❌

⏲️  Total pipeline duration: 01mn21s 

| Step                                                                  | Result   |
|:----------------------------------------------------------------------|:---------|
| Validate airbyte-integrations/connectors/source-hubspot/metadata.yaml | ✅        |
| Connector version semver check                                        | ✅        |
| Connector version increment check                                     | ✅        |
| QA checks                                                             | ✅        |
| Code format checks                                                    | ✅        |
| Connector package install                                             | ✅        |
| Build source-hubspot docker image for platform linux/x86_64           | ✅        |
| Unit tests                                                            | ❌        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/arsenlosenko_27799-source-hubspot-owners-archived-stream/1691582792/9e21e84bea6764a695bb6c0804bcfcd166b35141/source-hubspot/1.4.0/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:9e21e84bea6764a695bb6c0804bcfcd166b35141)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connector_ops/connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-hubspot test
```

**Comment by octavia-squidington-iii** [comment]
##  source-hubspot test report (commit [`9e21e84bea`](https://github.com/airbytehq/airbyte/pull/29249/commits/9e21e84bea6764a695bb6c0804bcfcd166b35141)) - ❌

⏲️  Total pipeline duration: 01mn27s 

| Step                                                                  | Result   |
|:----------------------------------------------------------------------|:---------|
| Validate airbyte-integrations/connectors/source-hubspot/metadata.yaml | ✅        |
| Connector version semver check                                        | ✅        |
| Connector version increment check                                     | ✅        |
| QA checks                                                             | ✅        |
| Code format checks                                                    | ✅        |
| Connector package install                                             | ✅        |
| Build source-hubspot docker image for platform linux/x86_64           | ✅        |
| Unit tests                                                            | ❌        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/arsenlosenko_27799-source-hubspot-owners-archived-stream/1691583005/9e21e84bea6764a695bb6c0804bcfcd166b35141/source-hubspot/1.4.0/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:9e21e84bea6764a695bb6c0804bcfcd166b35141)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connector_ops/connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-hubspot test
```

**Comment by octavia-squidington-iii** [comment]
##  source-hubspot test report (commit [`17ad753b39`](https://github.com/airbytehq/airbyte/pull/29249/commits/17ad753b39e7d0a47e6c4680261342a125d28548)) - ❌

⏲️  Total pipeline duration: 01mn21s 

| Step                                                                  | Result   |
|:----------------------------------------------------------------------|:---------|
| Validate airbyte-integrations/connectors/source-hubspot/metadata.yaml | ✅        |
| Connector version semver check                                        | ✅        |
| Connector version increment check                                     | ✅        |
| QA checks                                                             | ✅        |
| Code format checks                                                    | ✅        |
| Connector package install                                             | ✅        |
| Build source-hubspot docker image for platform linux/x86_64           | ✅        |
| Unit tests                                                            | ❌        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/arsenlosenko_27799-source-hubspot-owners-archived-stream/1691583636/17ad753b39e7d0a47e6c4680261342a125d28548/source-hubspot/1.4.0/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:17ad753b39e7d0a47e6c4680261342a125d28548)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connector_ops/connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-hubspot test
```

**Comment by octavia-squidington-iii** [comment]
##  source-hubspot test report (commit [`cff64785f6`](https://github.com/airbytehq/airbyte/pull/29249/commits/cff64785f6759a24d8a2cf01a9dd8a3be5c6aac9)) - ✅

⏲️  Total pipeline duration: 12mn04s 

| Step                                                                  | Result   |
|:----------------------------------------------------------------------|:---------|
| Validate airbyte-integrations/connectors/source-hubspot/metadata.yaml | ✅        |
| Connector version semver check                                        | ✅        |
| Connector version increment check                                     | ✅        |
| QA checks                                                             | ✅        |
| Code format checks                                                    | ✅        |
| Connector package install                                             | ✅        |
| Build source-hubspot docker image for platform linux/x86_64           | ✅        |
| Unit tests                                                            | ✅        |
| Integration tests                                                     | ✅        |
| Acceptance tests                                                      | ✅        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/arsenlosenko_27799-source-hubspot-owners-archived-stream/1691590285/cff64785f6759a24d8a2cf01a9dd8a3be5c6aac9/source-hubspot/1.4.0/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:cff64785f6759a24d8a2cf01a9dd8a3be5c6aac9)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connector_ops/connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-hubspot test
```

File: airbyte-integrations/connectors/source-hubspot/source_hubspot/streams.py [hunk 1]
```diff
@@ -1700,6 +1699,27 @@ class Owners(ClientSideIncrementalStream):
...
+    scopes = {"crm.objects.owners.read"}
+
+    def request_params(
+        self,
+        stream_state: Mapping[str, Any],
+        stream_slice: Mapping[str, Any] = None,
+        next_page_token: Mapping[str, Any] = None,
+    ) -> MutableMapping[str, Any]:
+        params = super().request_params(stream_state, stream_slice, next_page_token)
+        params.update({"archived": "true"})
```
Comment (bazarnov) [inline airbyte-integrations/connectors/source-hubspot/source_hubspot/streams.py:1719]: Let's make this explicitly added to the `params`: 

```python
params["archived"] = "true"
```

**Review by bazarnov — APPROVED** [review]

**Comment by octavia-squidington-iii** [comment]
##  source-hubspot test report (commit [`987ce3de6c`](https://github.com/airbytehq/airbyte/pull/29249/commits/987ce3de6ca5c02b1ddb3e151709932710cf2ba6)) - ✅

⏲️  Total pipeline duration: 12mn04s 

| Step                                                                  | Result   |
|:----------------------------------------------------------------------|:---------|
| Validate airbyte-integrations/connectors/source-hubspot/metadata.yaml | ✅        |
| Connector version semver check                                        | ✅        |
| Connector version increment check                                     | ✅        |
| QA checks                                                             | ✅        |
| Code format checks                                                    | ✅        |
| Connector package install                                             | ✅        |
| Build source-hubspot docker image for platform linux/x86_64           | ✅        |
| Unit tests                                                            | ✅        |
| Integration tests                                                     | ✅        |
| Acceptance tests                                                      | ✅        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/arsenlosenko_27799-source-hubspot-owners-archived-stream/1691665278/987ce3de6ca5c02b1ddb3e151709932710cf2ba6/source-hubspot/1.4.0/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:987ce3de6ca5c02b1ddb3e151709932710cf2ba6)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connector_ops/connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-hubspot test
```

File: airbyte-integrations/connectors/source-hubspot/acceptance-test-config.yml [hunk 2]
```diff
@@ -41,6 +41,8 @@ acceptance_tests:
             bypass_reason: Unable to populate
           - name: deals_archived
             bypass_reason: Unable to populate
+          - name: owners_archived
+            bypass_reason: unable to populate
```
Comment (lazebnyi) [inline airbyte-integrations/connectors/source-hubspot/acceptance-test-config.yml:45]: why this can not be populated?

**Comment by octavia-squidington-iii** [comment]
##  source-hubspot test report (commit [`fad0a5aa33`](https://github.com/airbytehq/airbyte/pull/29249/commits/fad0a5aa33f9feb70b965467fdb951f7b7c1152e)) - ✅

⏲️  Total pipeline duration: 11mn26s 

| Step                                                                  | Result   |
|:----------------------------------------------------------------------|:---------|
| Validate airbyte-integrations/connectors/source-hubspot/metadata.yaml | ✅        |
| Connector version semver check                                        | ✅        |
| Connector version increment check                                     | ✅        |
| QA checks                                                             | ✅        |
| Code format checks                                                    | ✅        |
| Connector package install                                             | ✅        |
| Build source-hubspot docker image for platform linux/x86_64           | ✅        |
| Unit tests                                                            | ✅        |
| Integration tests                                                     | ✅        |
| Acceptance tests                                                      | ✅        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/arsenlosenko_27799-source-hubspot-owners-archived-stream/1691744352/fad0a5aa33f9feb70b965467fdb951f7b7c1152e/source-hubspot/1.4.0/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:fad0a5aa33f9feb70b965467fdb951f7b7c1152e)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connector_ops/connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-hubspot test
```

**Comment by octavia-squidington-iii** [comment]
##  source-hubspot test report (commit [`8abd727a51`](https://github.com/airbytehq/airbyte/pull/29249/commits/8abd727a512c64dcb969828acbf79d4c88b67d1f)) - ✅

⏲️  Total pipeline duration: 10mn57s 

| Step                                                                  | Result   |
|:----------------------------------------------------------------------|:---------|
| Validate airbyte-integrations/connectors/source-hubspot/metadata.yaml | ✅        |
| Connector version semver check                                        | ✅        |
| Connector version increment check                                     | ✅        |
| QA checks                                                             | ✅        |
| Code format checks                                                    | ✅        |
| Connector package install                                             | ✅        |
| Build source-hubspot docker image for platform linux/x86_64           | ✅        |
| Unit tests                                                            | ✅        |
| Integration tests                                                     | ✅        |
| Acceptance tests                                                      | ✅        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/arsenlosenko_27799-source-hubspot-owners-archived-stream/1691754432/8abd727a512c64dcb969828acbf79d4c88b67d1f/source-hubspot/1.4.0/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:8abd727a512c64dcb969828acbf79d4c88b67d1f)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connector_ops/connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-hubspot test
```

File: airbyte-integrations/connectors/source-hubspot/acceptance-test-config.yml (hunk 2 above)
Comment (arsenlosenko) [reply airbyte-integrations/connectors/source-hubspot/acceptance-test-config.yml:45]: right now it was not possible to archive any owners in our test account from UI or API endpoint, so this stream would be bypassed for now

**Comment by octavia-squidington-iii** [comment]
##  source-hubspot test report (commit [`a52923bb81`](https://github.com/airbytehq/airbyte/pull/29249/commits/a52923bb8102c6b6252448e2f6a83bb2687ee9b2)) - ✅

⏲️  Total pipeline duration: 12mn08s 

| Step                                                                  | Result   |
|:----------------------------------------------------------------------|:---------|
| Validate airbyte-integrations/connectors/source-hubspot/metadata.yaml | ✅        |
| Connector version semver check                                        | ✅        |
| Connector version increment check                                     | ✅        |
| QA checks                                                             | ✅        |
| Code format checks                                                    | ✅        |
| Connector package install                                             | ✅        |
| Build source-hubspot docker image for platform linux/x86_64           | ✅        |
| Unit tests                                                            | ✅        |
| Integration tests                                                     | ✅        |
| Acceptance tests                                                      | ✅        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/arsenlosenko_27799-source-hubspot-owners-archived-stream/1691768858/a52923bb8102c6b6252448e2f6a83bb2687ee9b2/source-hubspot/1.4.0/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:a52923bb8102c6b6252448e2f6a83bb2687ee9b2)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connector_ops/connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-hubspot test
```

**Pull request merged**